lib/msf/ui/console/table_print/custom_color_styler.rb

```diff
@@ -0,0 +1,39 @@
+# -*- coding: binary -*-
+
+module Msf
+  module Ui
+    module Console
+      module TablePrint
+        class CustomColorStyler
+          RESET_COLOR = '%clr'.freeze
+          # @param [Hash<String, String>] opts A hash of a String to find, and what color to use
+          # For example: opts = { 'abc' => '%grn' }
+          def initialize(opts)
+            @highlight_terms = {}
+            opts.each { |term, color| @highlight_terms[term] = color }
+          end
+
+          def style(value)
```

Comment:
I posted this in the main thread, but commenting here for visibility

![image](https://user-images.githubusercontent.com/60357436/168803961-2bf6bd27-03b9-4b37-b2c0-a1dfba2516e9.png)

Breakpoint:
```
[5] pry(#<Msf::Ui::Console::TablePrint::CustomColorStyler>)> whereami 4

From: /Users/adfoster/Documents/code/metasploit-framework/lib/msf/ui/console/table_print/custom_color_styler.rb:20 Msf::Ui::Console::TablePrint::CustomColorStyler#style:

    16:           def style(value)
    17: 
    18:             if value.include?('Open')
    19:               require 'pry'; binding.pry
 => 20:               puts value
    21:             end
    22: 
    23:             if @highlight_terms.key?(value)
    24:               return "#{@highlight_terms[value]}#{value}#{RESET_COLOR}"

[6] pry(#<Msf::Ui::Console::TablePrint::CustomColorStyler>)> @highlight_terms
=> {"Unix"=>"%red", "Android"=>"%red", "Linux"=>"%red", "BSD"=>"%red", "Solaris"=>"%red", "OpenBSD"=>"%red", "OSX"=>"%red", "Windows"=>"%red"}
```

We may want to make use of [maximal munch](https://en.wikipedia.org/wiki/Maximal_munch) here